libraries/ui/src/ui/types/ContextAwareProfile.h

```diff
@@ -48,9 +78,13 @@ class ContextAwareProfile : public ContextAwareProfileParent {
 
     ContextAwareProfile(QQmlContext* parent);
     QQmlContext* _context{ nullptr };
-    bool _cachedValue{ false };
-    quint64 _cacheExpiry{ 0 };
-    constexpr static quint64 MAX_CACHE_AGE = MSECS_PER_SECOND;
+    QAtomicInt _isRestricted{ 0 };
```

Comment:
couldn't you use an `std::atomic<bool>` here?

Reply:
For this exercise I tried to de-STL my code as much as possible, using the Qt versions of code when both were available.

Reply:
it's actually probably in our best interest to decouple ourselves from Qt as much as possible!  we didn't always adhere to this rule (as you can tell since the code is full of Qt data types) but near the end we tried to replace things with STL as we went

Reply:
If you would prefer this changed to an atomic<bool> then that could be done, there's an awful lot of other Qt'isms in this code though and I don't really expect them all to be changed over.  I'm also not expecting this to turn into a debate of STL vs Qt (of which my opinion has been changing multiple times over the last few weeks)

Reply:
Yes, I'm not sure if it's worthwhile at this point for us to spend a lot of time on being independent from Qt. We're a much smaller project than hifi was, converting code risks introducing bugs without improving functionality, and it makes it harder to examine the code's history if you end up with hifi on one side and STL on the other.

Writing new code based on the STL in places where this is possible without having to convert a bunch of other stuff is probably a good idea, though.

Reply:
[off-topic] if it helps, my current position is:
- I trust QDeadlockTimer/QElapsed timer much more than the STL versions (the STL "high resolution" objects HiFi uses in the existing UDT code are deprecated as meaningless)
- the STL map<> objects are better than QMap<> when one needs a custom comparator (e.g. case-insensitive) or need std::move support, neither of which Qt does.

Reply:
my original point was also that this could be a bool instead of an int, and I don’t think there’s a QAtomicBool

Reply:
Answer is the same as before, If you would prefer this changed to an atomic then that could be done.